python/python_fastapi_server/routers/edr.py

```diff
@@ -615,14 +636,31 @@ async def rest_get_edr_collections(request: Request):
     response_model=Collection,
     response_model_exclude_none=True,
 )
-async def rest_get_edr_collection_by_id(collection_name: str):
+async def rest_get_edr_collection_by_id(collection_name: str, response: Response):
     """
     GET Returns collection information for given collection id
     """
-    collection = await get_collectioninfo_for_id(collection_name)
+    collection, expires = await get_collectioninfo_for_id(collection_name)
+    response.headers["cache-control"] = generate_max_age(expires)
     return collection
 
 
+def get_ttl_from_adaguc_call(headers):
```

Comment:
Please describe that it is reading the cache headers from adaguc

Reply:
Done